crates/iceberg/src/scan.rs

```diff
@@ -163,24 +164,32 @@ impl TableScan {
 
         Ok(iter(file_scan_tasks).boxed())
     }
+
+    pub async fn execute(
+        &self,
+        batch_size: Option<usize>,
```

Comment:
This should be put in `TableScanBuilder` rather than just passing as function argument.

Reply:
I can do this, but I wasn't sure that made sense, since `batch_size` might be only relevant for `TableScanBuilder`'s whose `to_arrow()` method get called? Will we have other methods on here equivalent to `to_arrow()` but returning different types?

Reply:
Yes, you are right. `TableScan` is used to define scan related args, such as filter, snapshot, while `batch_size` is mostly execution related, so it's appropriate to associate it with `ArrowReader`. The problem is with this api is that it's not extensible, since we need to break it every time we add a new config option for `ArrowReader`. How about removing this api, but passing `TableScan` to `ArrowReaderBuilder`?

Reply:
I've done this now